tests/ut_api_hmac.cpp

```diff
@@ -0,0 +1,17 @@
+
+#include <api/hmac.h>
+
+#include "gtest/gtest.h"
+using namespace testing;
+
+TEST(api_hmac, simple_digest) {
+
+	char digest[256];
+	unsigned char *data = (unsigned char *)"Test";
+	const char *secretkey = "secret";
+	size_t datalen = 4;
+
+	vz::hmac_sha1(digest, data, datalen, secretkey);
```

Comment:
if you'd change the signature to const unsigned char* and len you could use e.g.
"secret\0s" (hex: 7365637265740073) and should get the digest:
"79b538ca43bf49b46249a4e4ad0184c42fad4790"

Reply:
updated.